### PR TITLE
Domains: Reduxify Domains and remove Flux Store

### PR DESCRIPTION
**Pull request by aidvu**
- Rewrite the `*Data` components into a single component
- Remove DomainsStore
- Completely move `domains` to Redux (this doesn't include all the endpoints, only `/domains`)
- Update components
- Update tracking
- Clean code

----------------

This one was a pain because the data format for  `domains` from the `DomainsStore` isn't the same as the one from the Redux domains selectors.

Had to go through all of the components and update the format and how the data is used. Also updated helper functions that expected the `DomainsStore` format.

All the `*Data` wrappers are not needed in my opinion, and could've been written in a single class. Hopefully, I'll have time to cleanup some more of the Flux stuff before the next Hack Week.

-----------------

Test... This touches all of domain management. I'd suggest:
- Calypso -> Domains
- Check every domain type, Registration, Mapping, Transfer, Redirect, wp.com sudomain
- Open every nav item, then refresh, make sure all works even on reloads
- Test every action
- Have the Network tab open, monitor API calls (no duplicate `/domains` calls)

### CODE REVIEW

**Review by hambai — APPROVED** [review]
OK, so I went through all the flows I could think of and I'd say it works great 👍 . Couldn't find any regression and I'm so happy we're not making duplicate requests now :D We still do for GApps and the rest of the Flux stuff I guess, but it should be easy to reduxify those too now. I'd be happy if more people weigh in this PR.

**Comment by blowery** [comment]
@aidvu couple little merge conflicts to clean up, but this looks good

File: client/components/data/domain-management/index.jsx [hunk 1]
```diff
@@ -1,101 +1,240 @@
...
+		};
+	},
+	( dispatch, { selectedDomainName } ) => {
+		const googleAppsUsersFetcher = selectedDomainName
+			? () => fetchByDomain( selectedDomainName )
+			: siteId => fetchBySiteId( siteId );
 
-export default connect( mapStateToProps )( DomainManagementData );
+		return {
+			fetchGoogleAppsUsers: siteId => dispatch( googleAppsUsersFetcher( siteId ) ),
```
Comment (blowery) [inline client/components/data/domain-management/index.jsx]: i think this is going to be a new function on every run of `connect`, which means the component will rerender...

@jsnajdr do I have that right?

**Review by alisterscott — APPROVED** [review]
Needs a rebase but LGTM

**Comment by aidvu** [comment]
Merging on Monday when back from WCEU.

**Comment by dmsnell** [comment]
**Build Size Diff**
http://iscalypsofastyet.com/branch?branch=update/reduxify-domains
```
Delta:
chunk                                                             stat_size           parsed_size           gzip_size
async-load-my-sites-current-site-domain-warnings~domains~plans         +0 B                  +0 B                +1 B  (+0.0%)
build                                                               -2636 B  (-0.1%)      -1675 B  (-0.1%)     -829 B  (-0.3%)
checkout                                                               +0 B                  +0 B                +1 B  (+0.0%)
checkout~domains~jetpack-connect~plans~purchases~settings~signup    -5757 B  (-7.1%)      -2968 B  (-7.6%)     -574 B  (-6.0%)
domains                                                            -16961 B  (-3.0%)      -6290 B  (-2.3%)     -703 B  (-1.3%)
domains~signup                                                        +75 B  (+0.0%)        +38 B  (+0.0%)       +4 B  (+0.0%)
manifest                                                               +0 B                  +0 B                +2 B  (+0.0%)
```

**Comment by aidvu** [comment]
Regression with GApps fixed here: https://github.com/Automattic/wp-calypso/pull/25567

I did the GApps stuff a day after and forgot that `hasLoadedFromServer === ! isRequestingSiteDomains`. I just did a replace, and gapps broke.

**Pull request merged**